sdk/storage/azure-storage-file-share/azure/storage/fileshare/_directory_client.py

```diff
@@ -530,8 +493,7 @@ def rename_directory(
             process_storage_error(error)
 
     @distributed_trace
-    def list_directories_and_files(self, name_starts_with=None, **kwargs):
-        # type: (Optional[str], **Any) -> ItemPaged
+    def list_directories_and_files(self, name_starts_with: Optional[str] = None, **kwargs: Any) -> ItemPaged:
```

Comment:
This will be `ItemPaged[Union[DirectoryProperties, FileProperties]]` 
Then be careful of whether it needs to be in "" and imported in `TYPE_CHECKING` or not 😄